### PR TITLE
Pr/list retrieve post

### PR DESCRIPTION
**Pull request by MethodenMann**
Main goal of this PR:
- Add helper to fetch entities via /search POST request with possibility to overwrite endpoint and method
- Add entity count helper 
- Use these helpers in entity list
- clean up fetch options and add better comments.

some of these changes serve as preparation to send selections as query to the server.

Also done:
- fix some problems introduced with type mapping refactoring (#600)
- fix bug where invisible search form fields where not initialized correctly
- transform search form values by type and not by object analysis (and keep fallback for values that are not in the form)

### CODE REVIEW

File: packages/tocco-util/src/mockData/entities.js [hunk 1]
```diff
@@ -160,19 +185,33 @@ const createEntityResponse = (entityName, entityStore) => {
   }
 }
 
+const extractUrlParams = url => () => {
+  return {
```
Comment (rzueger) [inline packages/tocco-util/src/mockData/entities.js]: can be written without `return`

File: packages/entity-list/src/util/api/fetchOptions.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  } else {
+    const objectBased = objectTransformFallback(value, key)
+    if (objectBased) return objectBased
+  }
+
+  return {[key]: {value}}
+}
+
+export const getFetchOptionsFromSearchForm = (searchFormValues, formFields = {}) => {
```
Comment (rzueger) [inline packages/entity-list/src/util/api/fetchOptions.js]: can be written without `return`

File: packages/entity-list/src/modules/list/sagas.js [hunk 3]
```diff
@@ -112,26 +119,27 @@ export function* setSorting() {
 }
 
 export function* getSearchFilter(inputSearchFilters, searchInputsFilters) {
-  const filters = yield call(_union, inputSearchFilters, searchInputsFilters)
-  return yield call(_join, filters, ',')
+  return yield call(_union, inputSearchFilters, searchInputsFilters)
```
Comment (rzueger) [inline packages/entity-list/src/modules/list/sagas.js]: can be written without `return`

File: packages/app-extensions/src/rest/helpers.js [hunk 4]
```diff
@@ -85,34 +106,66 @@ export const defaultFormTransformer = json => (json.form)
...
+    ...(Array.isArray(relations) ? {'relations': relations.length === 0 ? '!' : relations} : {}),
+    ...(filters ? {'filter': filters} : {})
   }
+)
 
-  if (limit) {
-    params._limit = limit
-    if (page) {
-      params._offset = (page - 1) * limit
+export const queryObjectToUrlParams = queryObject => (
```
Comment (rzueger) [inline packages/app-extensions/src/rest/helpers.js]: can be written without `(`...

File: packages/app-extensions/src/rest/helpers.js [hunk 5]
```diff
@@ -15,31 +17,50 @@ import {call} from 'redux-saga/effects'
...
- * - query {String} TQL query
+ * @param options {Object} see 'buildEntityQueryObject' function
  * @param transformer {function} Function to directly manipulate the result. By default returns data attribute
+ * @param endpoint {String} To overwrite default endpoint entities/entityName/search
+ * @param method {String} To overwrite default http method 'POST'
  */
-export function* fetchEntities(entityName, options, transformer = defaultEntityTransformer) {
-  const queryParams = yield call(buildParams, options)
-  const resp = yield call(requestSaga, `entities/${entityName}`, {queryParams})
+export function* fetchEntities(entityName, options, transformer = defaultEntityTransformer, endpoint, method = 'POST') {
```
Comment (rzueger) [inline packages/app-extensions/src/rest/helpers.js]: - quite a lot method parameters. shouldn't become any more than 5, IMO
- or is there a way to reduce the number of parameters?
- can we rename `options` to something a little bit more specific?

File: packages/entity-list/src/modules/list/sagas.js (hunk 3 above)
Comment (MethodenMann) [reply packages/entity-list/src/modules/list/sagas.js]: generator function....

File: packages/entity-list/src/util/api/forms.js [hunk 6]
```diff
@@ -59,3 +59,20 @@ export const getFields = formDefinition => {
 
   return _uniq(fields)
 }
+
+export const getFormFieldFlat = formDefinition => {
+  return searchChildren(formDefinition)
```
Comment (rzueger) [inline packages/entity-list/src/util/api/forms.js]: (superfluous `return`)

File: packages/entity-list/src/util/api/forms.js [hunk 7]
```diff
@@ -59,3 +59,20 @@ export const getFields = formDefinition => {
...
+}
+
+const searchChildren = node => {
+  if (node.componentType === 'field-set') {
+    return node.children.reduce((acc, value) => ({...acc, [value.id]: value.dataType}), {})
+  } else {
+    if (node.children) {
```
Comment (rzueger) [inline packages/entity-list/src/util/api/forms.js]: (could be simplified to `else if ...`)

**Pull request merged**